extensions/mapstore/src/test/java/com/hazelcast/mapstore/mongodb/MongoGenericMapStoreTest.java

```diff
@@ -0,0 +1,185 @@
+        tableName = randomName();
+        createCollection(tableName);
+        database.getCollection(tableName).insertOne(new Document("id", 1).append("name", "name-1"));
+
+        MapConfig mapConfig = new MapConfig(tableName);
+        MapStoreConfig mapStoreConfig = new MapStoreConfig();
+        mapStoreConfig.setClassName(GenericMapStore.class.getName());
+        mapStoreConfig.setProperty("data-link-ref", TEST_DATABASE_REF);
+        mapStoreConfig.setProperty("type-name", "org.example.Person");
+        mapConfig.setMapStoreConfig(mapStoreConfig);
+        instance().getConfig().addMapConfig(mapConfig);
+    }
+
+    private void createCollection(String collectionName) {
+        CreateCollectionOptions options = new CreateCollectionOptions();
+        ValidationOptions validationOptions = new ValidationOptions();
+        validationOptions.validator(BsonDocument.parse(
+                "{\n" +
+                        "    $jsonSchema: {\n" +
+                        "      bsonType: \"object\",\n" +
+                        "      title: \"Object Validation\",\n" +
+                        "      properties: {" +
+                        "        \"id\": { \"bsonType\": \"int\" },\n" +
+                        "        \"name\": { \"bsonType\": \"string\" }\n" +
+                        "      }\n" +
+                        "    }\n" +
+                        "  }\n"
+        ));
+        options.validationOptions(validationOptions);
+        database.createCollection(collectionName, options);
+    }
+
+    @Test
+    public void testGet() {
+        HazelcastInstance client = client();
+        IMap<Integer, Person> map = client.getMap(tableName);
+
+        Person p = map.get(0);
+        assertThat(p.getId()).isZero();
+        assertThat(p.getName()).isEqualTo("name-0");
+    }
+
+    @Test
+    public void testPut() {
+        HazelcastInstance client = client();
+        IMap<Integer, Person> map = client.getMap(tableName);
+
+        map.put(42, new Person(42, "name-42"));
+
+        assertMongoRowsAnyOrder(tableName,
+                new Person(0, "name-0"),
+                new Person(42, "name-42")
+        );
+    }
+
+    @Test
+    public void testPutWhenExists() {
+        HazelcastInstance client = client();
+        IMap<Integer, Person> map = client.getMap(tableName);
+
+        assertMongoRowsAnyOrder(tableName,
+                new Person(0, "name-0")
+        );
+
+        map.put(0, new Person(0, "updated"));
+
+        assertMongoRowsAnyOrder(tableName,
+                new Person(0, "updated")
+        );
+    }
+
+    @Test
+    public void testRemove() {
+        HazelcastInstance client = client();
+        IMap<Integer, Person> map = client.getMap(tableName);
+
+        Assertions.assertThat(database.getCollection(tableName).countDocuments()).isEqualTo(1);
+
+        map.remove(0);
+
+        Assertions.assertThat(database.getCollection(tableName).countDocuments()).isEqualTo(0);
```

Comment:
You can use shorter line
Assertions.assertThat(database.getCollection(tableName).countDocuments()).isZero(); 
instead of 
Assertions.assertThat(database.getCollection(tableName).countDocuments()).isEqualTo(0);